### PR TITLE
feature/1584 - Add PHNL - Daniel K Inouye International Airport

### LINKED ISSUE #1584: Add PHNL - Daniel K Inouye International Airport
Time for some island airports!
Starting with PHNL - Daniel K Inouye International Airport

### PR DESCRIPTION
**Pull request by JPmAn24**
Resolves #1584 

The purpose of this pull request is to add PHNL - Daniel K Inouye International Airport

### CODE REVIEW

**Comment by MichaelDMcCracken** [comment]
Does anyone care if I revive this?

**Comment by MichaelDMcCracken** [comment]
@erikquinn @Fechulo 

Can someone update this branch to 6.26.0 or 6.27.0? I'm keep screwing it up when I try to do it locally.

File: assets/airports/phnl.json [hunk 1]
```diff
@@ -0,0 +1,6641 @@
...
+            "draw": [
+                ["CARRP", "OPACA", "HNL"],
+                ["KATHS", "OPACA", "BINJO", "CHOKO"]
+            ]
+        },
+        "SAKKI5": {
+            "icao": "SAKKI5",
+            "name": "Sakki Five",
+            "entryPoints": {
+                "CHAIN": ["CHAIN", "SOBOW", "SERAH"],
```
Comment (Fechulo) [inline assets/airports/phnl.json]: The 13DME arc returns!

File: assets/airports/phnl.json [hunk 2]
```diff
@@ -0,0 +1,6641 @@
...
+                ["DOVRR", "DAFRE", "JORDA", "SUCTU", "SAKKI", "SECIL"],
+                ["HOKLA", "FIRES", "CRISI", "JORDA"],
+                ["LNY", "GRAMY", "SOBOW", "SERAH", "SAKKI", "JULLE"],
+                ["CHAIN", "SOBOW"]
+            ]
+        },
+        "SHLAE1": {
+            "icao": "SHLAE1",
+            "name": "Shlae One",
+            "entryPoints": {
```
Comment (Fechulo) [inline assets/airports/phnl.json]: Missing LNY transition

**Review by Fechulo — CHANGES_REQUESTED** [review]
Missing fix definitions
image

If you have the time and energy, it'd be nice to modify the terrain to match the video map. This is fairly straight forward to do using qGIS with tracing on

File: assets/airports/phnl.json [hunk 3]
```diff
@@ -0,0 +1,6641 @@
...
+                "PHNL8R": ["_RW8R1DME", "#150"],
+                "PHNL26L": ["_RW26L1DME", "#180"],
+                "PHNL26R": ["_RW26R1DME", "#180"]
+            },
+            "body": ["KEOLA"],
+            "exitPoints": {
+                "KATHS": ["LEANE", "NONNI", "KATHS"],
+                "LIH": ["GRAIL", "NAPUA", "LIH"],
+                "LILIA": ["LILIA"],
+                "NONNI": ["NONNI"],
```
[the remark this answers is not in the document]
Comment (MichaelDMcCracken) [reply assets/airports/phnl.json]: - [x]

File: assets/airports/phnl.json [hunk 4]
```diff
@@ -0,0 +1,6641 @@
...
+                ["DENNS", "DLIMA", "JOELE", "BAMBO", "HISPO", "BOWSZ", "BITTA"],
+                ["APACK", "MSONA"],
+                ["INOYI", "HUBAP"]
+            ]
+        },
+        "JULLE5": {
+            "icao": "JULLE5",
+            "name": "Julle Five",
+            "entryPoints": {
+                "CHAIN": ["CHAIN", "SOBOW", "SERAH", "SAKKI"],
```
[the remark this answers is not in the document]
Comment (MichaelDMcCracken) [reply assets/airports/phnl.json]: I was being lazy.... but I have added this nonetheless

**Comment by MichaelDMcCracken** [comment]
@Fechulo 
Requested changes have been pushed.

**Review by Fechulo — APPROVED** [review]

**Review by erikquinn — CHANGES_REQUESTED** [review]
@MichaelDMcCracken Applied this PR to the latest version of the sim, and there are some issues with the formatting of certain sections that are causing issues. Just need to address those so it works with the latest version 🙂

**Comment by MichaelDMcCracken** [comment]
> @MichaelDMcCracken Applied this PR to the latest version of the sim, and there are some issues with the formatting of certain sections that are causing issues. Just need to address those so it works with the latest version 🙂

So I updated this locally (the restricted areas need to be defined as "poly" instead of "coordinates"), but totally goobered things up with git. Might be easier if you make the edit yourself, because my local branch is toast ...

image

**Comment by erikquinn** [comment]
It's an easy fix. I sent you a message on slack :)

**Pull request merged**